modules/oaipmh-persistence/src/main/java/org/opencastproject/oaipmh/persistence/impl/AbstractOaiPmhDatabase.java

```diff
@@ -173,71 +147,27 @@ public void updateEntity(OaiPmhEntity entity, MediaPackage mediaPackage,
         continue;
       }
 
+      if (mpe.getMimeType() == null || !mpe.getMimeType().eq(MimeTypes.XML)) {
```

Comment:
Let's hope that the mime type of catalogs is always set correctly....

Reply:
The alternative way (not to check the mimetype) is trying to parse the xml document. I think it will be overkill.

Reply:
Just wonder whether "old" data has the mime type always set correctly... and you do parse the XML document just below (line 162) anyway, don't you?